native/swift/Sources/wordpress-api/LoginAPI.swift

```diff
@@ -10,8 +10,8 @@ import FoundationNetworking
 
 public final class WordPressLoginClient {
 
-    public protocol Authenticator {
-        func authenticate(url: URL, callbackURL: URL) async -> Result<URL, Error>
+    public protocol AuthenticatorProtocol {
+        func authenticate(url: URL, callbackURL: URL) async throws -> URL
```

Comment:
I used `Result` so that we can easily change the API to `throws(Error) -> URL`, but I forgot to do it...

Now that you are changing this, do you mind adding the error type here: `throws(Error)`? You'll probably need to undo your changes first because it should be more straightforward to update based on the original code, like changing `return result` to `return try result.get()` should be sufficient.

Reply:
Yep, I ran around and adjusted it – I also broke the error out into its own file since it's pretty big (and I suspect may get bigger)

Reply:
There was also some weird SwiftLint behaviour when I made the `throws` typed – it was yelling about nested type depth. 

Updating `SwiftLint` fixed it